File: codechef/2021 - August Long/array filling.py
for _ in range(int(input())):
    n,m = map(int,input().split())
    oper = {}
    for __ in range(m):
        x,y = map(int,input().split())
        if y in oper:
            if x > oper[y]:
                oper[y] = x
        else:
            oper[y] = x

    key,val = list(oper.keys()),list(oper.values())
    oper = {key[val.index(i)] : i for i in sorted(val,reverse=True)}
    
    sum = 0
    pos = n
    for i in oper:
        spots = pos - (pos//i)
        pos = pos//i
        sum += oper[i] * spots
    
    print(sum)

File: codechef/2021 - August Long/array filling.cpp
#include <bits/stdc++.h>
using namespace std;

void sort(int x[], int y[], int n) {
    int pos,max;
    for (int i = 0; i < n; i++) {
        pos = i; max = x[i];
        for (int j=i+1; j<n; j++){
            if (x[j] > x[i])
                pos = j; max =x[j];
        }
        int temp;
        temp = x[i]; x[i] = x[pos]; x[pos] = temp;
        temp = y[i]; y[i] = y[pos]; y[pos] = temp;
    }
}

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int t;
    cin >> t;
    for (int i=0; i<t; i++) {
        int n,m;
        cin >> n >> m;
        int x[m],y[m];
        for (int j=0; j<m; j++) {
            cin >> x[j] >> y[j];
        }
        sort(x,y,m);

        int sum = 0;
        int pos = n;
        for (int o=0; o<m; o++) {
            int spots = pos - (pos/y[o]);
            pos = pos/y[o];
            sum += x[o] * spots;
        }
        cout << sum << "\n";
    }
    return 0;
}